configure.ac

```diff
@@ -2613,6 +2613,15 @@ AS_CASE([$coroutine_type], [yes|''], [
         [aarch64-freebsd*], [
             coroutine_type=arm64
         ],
+        [riscv64-freebsd*], [
+            coroutine_type=riscv64
+        ],
+        [arm*-freebsd*], [
+            coroutine_type=arm32
+        ],
+        [powerpc-freebsd*], [
+            coroutine_type=ppc
```

Comment:
@ioquatix In principle, I could test it – I have hardware and FreeBSD is supposed to be supported. But no sooner than next week and provided the OS installs. (I do not imply the PR should wait for that, of course, I just mean I can assist with testing, be it needed.)

Reply:
I have ran `make test` - all the tests are passing.

Reply:
> I have ran `make test` - all the tests are passing.

@pkubaj There is one place where BE code was missing, could you check it, please? Perhaps *BSD gonna need it as well, though I am not sure: https://github.com/ruby/ruby/commit/e7c862aa6856e3b13536eb8514b7e37ee6c2c7fd

P. S. My updated for coroutines for Darwin PPC are here: https://github.com/ruby/ruby/commit/644c2629a89c857cb061241a05e09d97d9131b86
This is WIP, however. I will try to finalize it soon, at least `ppc32` part (`ppc64` is tricky since the build needs to be fixed for 10.5 first).